lcls_tools/common/image/fit.py

```diff
@@ -55,25 +77,70 @@ class ImageProjectionFit(ImageFit):
+        2.0, description="Fit amplitude to noise threshold for the fit"
+    )
+    beam_extent_n_stds: PositiveFloat = Field(
+        2.0,
+        description="Number of standard deviations on either side to use for the beam extent",
+    )
 
     def _fit_image(self, image: ndarray) -> ImageProjectionFitResult:
-        x_projection = np.array(np.sum(image, axis=0))
-        y_projection = np.array(np.sum(image, axis=1))
+        dimensions = ("x", "y")
+        fit_parameters = []
+        signal_to_noise_ratios = []
+        beam_extent = []
+
+        for axis, dim in enumerate(dimensions):
+            projection = np.array(np.sum(image, axis=axis))
+            x = np.arange(len(projection))
+            parameters = self.projection_fit_method(x, projection)
 
-        proj_fit = ProjectionFit(model=self.projection_fit_method)
+            signal_to_noise_ratios.append(gaussian.signal_to_noise(parameters))
 
-        x_parameters = proj_fit.fit_projection(x_projection)
-        y_parameters = proj_fit.fit_projection(y_projection)
+            # calculate the extent of the beam in the projection - scaled to the image size
+            beam_extent.append(gaussian.extent(parameters, self.beam_extent_n_stds))
+
+            # perform validation checks, modify parameters if checks fail
+            self._validate_parameters(
+                parameters, signal_to_noise_ratios, beam_extent, projection, dim
+            )
+
+            fit_parameters.append(parameters)
 
         result = ImageProjectionFitResult(
-            centroid=[x_parameters["mean"], y_parameters["mean"]],
-            rms_size=[x_parameters["sigma"], y_parameters["sigma"]],
+            centroid=[ele["mean"] for ele in fit_parameters],
+            rms_size=[ele["sigma"] for ele in fit_parameters],
             total_intensity=image.sum(),
-            x_projection_fit_parameters=x_parameters,
-            y_projection_fit_parameters=y_parameters,
+            projection_fit_parameters=fit_parameters,
             image=image,
             projection_fit_method=self.projection_fit_method,
+            curve=self.curve,
+            signal_to_noise_ratio=signal_to_noise_ratios,
+            beam_extent=beam_extent,
         )
 
         return result
+
+    def _validate_parameters(
+        self, parameters, signal_to_noise_ratios, beam_extent, projection, dim
+    ):
+        # if the amplitude of the the fit is smaller than noise then reject
+        # moving this into a validate function to clean it up.
+        if signal_to_noise_ratios[-1] < self.signal_to_noise_threshold:
+            for name in parameters.keys():
+                parameters[name] = np.nan
+
+            warnings.warn(f"Projection in {dim} had a low amplitude relative to noise")
+
+        # if the beam extent is outside the image then its off the screen etc. and fits cannot be trusted
+        if beam_extent[-1][0] < 0 or beam_extent[-1][1] > len(projection):
```

Comment:
@roussel-ryan 's comment, do this check on the single result before looking at the array. This would remove need for -1 indexing.